tenseal/tensors/ckksvector.cpp

```diff
@@ -174,6 +194,26 @@ CKKSVector& CKKSVector::sub_plain_inplace(vector<double> to_sub) {
     return *this;
 }
 
+CKKSVector CKKSVector::sub_plain(double to_sub) {
+    CKKSVector new_vector = *this;
+    new_vector.sub_plain_inplace(to_sub);
+
+    return new_vector;
+}
+
```

Comment:
It might help adding
```
template<typename T>
CKKSVector& CKKSVector::sub_plain_inplace_impl(T to_sub)
```

with the current `_plain_inplace` body.

The API can call this template method, after doing the specific checks for vector/double cases.

The same for add/multiply methods

Reply:
Cool! So there isn't any restriction on what T might be!